src/engine/enginebuffer.cpp

```diff
@@ -590,8 +591,16 @@ void EngineBuffer::slotControlSeek(double fractionalPos) {
 
 // WARNING: This method runs from SyncWorker and Engine Worker
 void EngineBuffer::slotControlSeekAbs(double playPosition) {
-    doSeekPlayPos(playPosition, SEEK_STANDARD);
-}
+    auto pMacro = m_pEngineMaster->m_pMacroRecording;
+    if (pMacro != nullptr) {
+        if (pMacro->deck == NO_DECK) {
+            pMacro->deck = this->getGroup();
```

Comment:
I think this allocates memory
Also, this needs to be atomic in case you press hotcue on 2 decks at the same time.

Reply:
I don't think we really have to worry about someone pressing it on two decks at the same time - he must be trying other intentionally break it and have extraordinary coordination.

As for allocation - this doesn't happen in the RT thread ;)

Reply:
@xerus2000 It is obvious that you have no experience with concurrency ;) Everything could and **will** happen. Even things that you could never imagine! Please don't argue and instead think about everything that could **possibly** happen.

Reply:
Use ChannelHandle to identify decks within the audio thread.

Reply:
Okay, the thread below discusses the concurrency issues :)
@Be-ing what do you mean? Where do I get this ChannelHandle from?

Reply:
ChannelHandleFactory

Reply:
ah, got it. But where do I get the actual id of the deck, except for inspecting the string?